### PR TITLE
endpoint: remove redundant doUpdateHostsFile() function

### PR DESCRIPTION
**Pull request by thaJeztah**
The second (sandbox) argument was unused, and it was only used in a single location, so we may as well inline the check.

### CODE REVIEW

**Comment by thaJeztah** [comment]
@arkodg @cpuguy83 PTAL

File: endpoint.go [hunk 1]
```diff
@@ -497,7 +497,7 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) (err error) {
 		n.getController().watchSvcRecord(ep)
 	}
 
-	if doUpdateHostsFile(n, sb) {
+	if !n.ingress && n.Name() != libnGWNetwork {
```
Comment (arkodg) [inline endpoint.go:500]:
```suggestion
      // Do not update hosts file with internal networks endpoint IP
	if !n.ingress && n.Name() != libnGWNetwork {
```

Comment (thaJeztah) [reply endpoint.go:501]: updated 👍

**Review by arkodg — APPROVED** [review]

**Comment by thaJeztah** [comment]
replaced by https://github.com/moby/moby/pull/42505

**Pull request closed without merging**